### PR TITLE
v3.2.0-b: cleaning, half-map FSCs, and relion-3.1 

### PR DESCRIPTION
**Pull request by michal-g**
### New Features
- introducing `cryodrgn_utils clean`, a new tool for removing extraneous output files from completed experiments (#297)
- `backproject_voxel` now produces half-maps and a half-map FSC (#329)
- creating `cryodrgn_utils fsc` from the `fsc` analysis script for calculating Fourier shell correlations between two `.mrc` volume files

### Addressing Issues and Bugs
- making `write_star` use RELION 3.1 format by default with optics groups generated from image size, pixel size, voltage, spherical aberration, and amplitude contrast; `--relion30` to use old format (#324)
- updating install setup to prevent use of Python 3.11 (see #306)
- `abinit_homo` now saves a `config.yaml` with a summary of parameters used, like `abinit_het`, `train_vae`, and `train_nn`
- fixing `filter_star` to accept tilt series as well (#335)
- fixing `affinity` bug in `analyze_landscape` (#345)

### Codebase Maintenance
- cleaner implementation of command-line interface
- updating unit tests for new APIs

### CODE REVIEW

**Comment by michal-g** [comment]
I added a second round of commits related solely to testing, with new tests added for `cryodrgn_utils fsc` and `cryodrgn_utils clean`. I also used the fixtures created for these tests to rewrite, consolidate, and expand existing tests for `abinit` and `train_nn`, which have been consolidated at `test_reconstruction`. This has also allowed us to introduce new features such as automatic teardown of resources once testing is completed.

Other existing tests, such as `test_pipeline` and `test_quick`, are also basically integration tests of the reconstruction model and downstream analyses and can be likewise refactored using these fixtures. I will leave this for a subsequent release, as we are in the middle of refactoring the manner in which these reconstruction models are invoked for `v4.0.0`.

Finally, `test_dataset` has been failing for a while as it was still expecting the second element of a batch returned by `torch.utils.data.DataLoader/cryodrgn.dataset.make_dataloader` to be an array in the SPA case, instead of the current behaviour, which is to return `None` (the second element is an array in the tilt-series case, compare e.g. `DataShuffler`). I have fixed this, although `test_eval_images` is still failing due to an outdated API and will need to also be fixed in an upcoming release.

**Comment by michal-g** [comment]
I've added another series of commits that create `cryodrgn_utils gen_mask` using [this skeleton script](https://github.com/ml-struct-bio/cryoem_tools/blob/main/gen_mask.py) as a starting point, with reparametrization in Angstroms and a cleaned up interface.

**Comment by michal-g** [comment]
Another set of commits here, spurred by needing a patch for #356 but also:
 - adding volume assets for use in tests of mask generation as well as of masking in FSC calculation
 - fixing issues in `cryodrgn_utils fsc` documentation
 - removing references to `scipy.ndimage.morphology` which is deprecated

**Comment by michal-g** [comment]
Another set of commits addressing #358; I've also improved the I/O interface in both `flip_hand` and `invert_contrast` so that:
- both commands still take an unnamed argument as `input`
- both can be run without `-o`, in which case the output file is the same as input with `_inverted` or `_flipped` appended
- both automatically create parent directories if present in `-o`

Also added more unit tests of both to cover the bug and this new behaviour!
